simtbx/nanoBragg/nanoBragg_crystal.py

```diff
@@ -9,17 +9,20 @@
 
 class NBcrystal(object):
 
-  def __init__(self):
-    ucell = (79.1, 79.1, 38.4, 90, 90, 90)
-    self.xtal_shape = "gauss"  # shapetype.Gauss
-    self.Ncells_abc = (10, 10, 10)
-    self.mos_spread_deg = 0
-    self.n_mos_domains = 1
-    self.thick_mm = 0.1
-    self.symbol = 'P43212'
-    self.miller_array = NBcrystal.dummie_Fhkl(ucell, self.symbol)
-    self.dxtbx_crystal = NBcrystal.dxtbx_crystal_from_ucell_and_symbol(ucell_tuple_Adeg=ucell,
-                                                                       symbol=(self.symbol))
+  def __init__(self, default=False):
```

Comment:
I thought python sometimes raises exceptions when creating objects whose properties are not initialized, at the least set them to None ? Some of the property methods might need to be updated, also can we keep the original behavior of defaulting to the default crystal   ?
```
def __init__(self, use_default_crystal=True): 
    self.xtal_shape = None  # string , one of the nanoBragg shape types, see below
    self.Ncells_abc = None  # 3 tuple defining mosaic domain size
    self.mos_spread_deg = None  # angular spread in degrees
    self.n_mos_domains = None  # number of domains 
    self.thick_mm = None  # crystal thickness used to scale the nanoBragg model 
    self.symbol = None  # space group symbol
    self.miller_array = None   # CCTBX miller array defining scattering amplitudes
    self.dxtbx_crystal = None     # dxtbx crystal model
    if use_default_crystal:
      ucell = (79.1, 79.1, 38.4, 90, 90, 90)
      self.xtal_shape = "gauss"  # shapetype.Gauss
      self.Ncells_abc = (10, 10, 10)
      self.mos_spread_deg = 0
      self.n_mos_domains = 1
      self.thick_mm = 0.1
      self.symbol = 'P43212'
      self.miller_array = NBcrystal.dummie_Fhkl(ucell, self.symbol)
      self.dxtbx_crystal = NBcrystal.dxtbx_crystal_from_ucell_and_symbol(ucell_tuple_Adeg=ucell,
                                                                         symbol=(self.symbol))
```

Also, I didnt think you would ever need to set self._miller_array directly  ?